File: src/main/java/net/guizhanss/ultimategenerators2/implementation/items/generators/EndlessGenerator.java
package net.guizhanss.ultimategenerators2.implementation.items.generators;

import javax.annotation.ParametersAreNonnullByDefault;

import org.bukkit.Location;
import org.bukkit.inventory.ItemStack;

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;
import io.github.thebusybiscuit.slimefun4.core.attributes.EnergyNetProvider;

import me.mrCookieSlime.CSCoreLibPlugin.Configuration.Config;

public class EndlessGenerator extends SlimefunItem implements EnergyNetProvider {
    @ParametersAreNonnullByDefault
    public EndlessGenerator(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(itemGroup, item, recipeType, recipe);
    }

    @Override
    @ParametersAreNonnullByDefault
    public int getGeneratedOutput(Location location, Config config) {
        return location.getBlock().getBlockPower() * 256;
    }

    @Override
    public int getCapacity() {
        return 256 * 15;
    }
}

File: src/main/kotlin/net/guizhanss/ultimategenerators2/implementation/items/generators/simple/EndlessGenerator.kt
@file:Suppress("DEPRECATION")

package net.guizhanss.ultimategenerators2.implementation.items.generators.simple

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import io.github.thebusybiscuit.slimefun4.core.attributes.EnergyNetProvider
import me.mrCookieSlime.CSCoreLibPlugin.Configuration.Config
import org.bukkit.Location
import org.bukkit.inventory.ItemStack

class EndlessGenerator(
    itemGroup: ItemGroup,
    item: SlimefunItemStack,
    recipeType: RecipeType,
    recipe: Array<ItemStack?>
) : SlimefunItem(itemGroup, item, recipeType, recipe), EnergyNetProvider {
    override fun getGeneratedOutput(location: Location, config: Config) = location.block.blockPower * 256

    override fun getCapacity() = 256 * 15
}
